src/services/EnsService.ts

```diff
@@ -23,68 +29,183 @@ function getEnsAddress (networkId: string) {
+    return this.isSupported
+      ? this.isRegistered(name).then((isRegistered) => !isRegistered)
+      : null
   }
 
-  owner (name: string): Promise<string> {
-    return this.ens.accountInfo(name).then(info => info.owner_key)
+  async owner (name: string): Promise<string> {
+    if (!this.isSupported(name)) return
+    const address = await this.tokenId(name)
+    return this.ensRegistrar.ownerOf(address)
   }
 
   manager (name: string): Promise<string> {
-    return this.ens.accountInfo(name).then(info => info.manager_key)
+    if (!this.isSupported(name)) return
+    return this.ens.name(name).getOwner()
   }
 
   tokenId (name: string): Promise<string> {
-    return null
-  }
-
-  record (name: string, key: string): Promise<RecordItem> {
-    return null
+    if (!this.isSupported(name)) return
+    const nameArray = name.split('.')
+    const label = nameArray[0]
+    const tokenID: string = labelhash(label).toString()
+    return Promise.resolve(tokenID)
+  }
+
+  // key: type.subtype -> 'address.eth','text.email'
+  async record (name: string, key: string): Promise<RecordItem> {
+    if (!this.isSupported(name)) return
+    const ensName = this.ens.name(name)
+    const keyArray = key.split('.')
+    const type = keyArray.length > 1 ? keyArray[0] : ''
+    const subtype = keyArray[keyArray.length - 1]
+    let value
+    if (type === 'address') {
+      value = await ensName.getAddress(subtype)
+    }
+    else {
+      value = await ensName.getText(subtype)
+    }
+    return await Promise.resolve({
+      key,
+      type,
+      subtype,
+      label: '',
+      value,
+      ttl: 0,
+    })
   }
 
   records (name: string, keys?: string[]): Promise<RecordItem[]> {
-    return null
-  }
-
-  addrs (name: string, keys?: string | string[]): Promise<RecordItemAddr[]> {
-    return null
-  }
-
-  addr (name: string): Promise<RecordItemAddr> {
-    return null
+    if (!this.isSupported(name)) return
+    if (!keys) keys = getTextRecordKeys().map((key) => `text.${key}`)
+    const requestArray: Array<Promise<RecordItem>> = []
+    keys.forEach((key) => requestArray.push(this.record(name, key)))
+    return Promise.all<RecordItem>(requestArray)
+  }
+
+  async addrs (
+    name: string,
+    keys?: string | string[]
+  ): Promise<RecordItemAddr[]> {
+    if (!this.isSupported(name)) return
+    if (!keys) {
+      keys = getAddrRecordKeys()
+    }
+    if (Array.isArray(keys)) {
+      const records = await this.records(
+        name,
+        keys.map((key) => `address.${key}`)
+      )
+      return records.map((record) => ({
+        ...record,
+        symbol: record.key.split('.')[1].toUpperCase(),
+      }))
+    }
+    else {
+      const record = await this.record(name, `address.${keys}`)
+      return [
+        {
+          ...record,
+          symbol: keys.toUpperCase(),
+        },
+      ]
+    }
+  }
+
+  async addr (name: string): Promise<RecordItemAddr> {
```

Comment:
The interface should be `addr(name: string, key: string)`.
Users can either pass in `coin_type` or `symbol` of the token.